src/js/tracing/gesturetracing.ts

```diff
@@ -0,0 +1,149 @@
+import { getCurrentHub } from '@sentry/core';
+import type { Breadcrumb, Hub } from '@sentry/types';
+import { logger } from '@sentry/utils';
+
+import { UI_ACTION } from './ops';
+import { ReactNativeTracing } from './reactnativetracing';
+
+export const DEFAULT_BREADCRUMB_CATEGORY = 'gesture';
+export const DEFAULT_BREADCRUMB_TYPE = 'user';
+
+export const GESTURE_POSTFIX_LENGTH = 'GestureHandler'.length;
+export const ACTION_GESTURE_FALLBACK = 'gesture';
+
+/**
+ * Internal interface following RNGH 2 Gesture Event API.
+ * We need to use this to avoid importing RNGH 2 types and depending on it.
+ * https://github.com/software-mansion/react-native-gesture-handler/blob/f0868f7ccf678c947ef65519ebf97ae149a10289/src/handlers/gestures/gesture.ts#L55
+ * @hidden
+ */
+type GestureEvent = Record<string, unknown>;
+
+/**
+ * Internal interface for RNGH 2 Gesture API.
+ * We need to use this to avoid importing RNGH 2 types and depending on it.
+ * https://github.com/software-mansion/react-native-gesture-handler/blob/2.9.0/src/handlers/gestures/gesture.ts#L120
+ * @hidden
+ */
+interface BaseGesture {
+  handlers?: {
+    onBegin?: (event: GestureEvent) => void;
+    onEnd?: (event: GestureEvent) => void;
+  };
+  handlerName: string;
+}
+
+interface GestureTracingOptions {
+  getCurrentHub: () => Hub;
+}
+
+/**
+ * Patches React Native Gesture Handler v2 Gesture to start a transaction on gesture begin with the appropriate label.
+ * Example: ShoppingCartScreen.dismissGesture
+ */
+export function traceGesture<GestureT>(
+  /**
+   * Label of the gesture to be used in transaction name.
+   * Example: dismissGesture
+   */
+  label: string,
+  gesture: GestureT,
+  options: Partial<GestureTracingOptions> = {},
+): GestureT {
+  const gestureCandidate = gesture as unknown as BaseGesture | undefined | null;
+  if (!gestureCandidate) {
+    logger.warn('[ReactNativeTracing] Gesture can not be undefined');
```

Comment:
L: Would make sense to use `[GestureTracing]` instead? so you know where it's coming from, not sure if the rest of the SDK does that.

Reply:
At least the `[ReactNativeTracing]` does it. Makes sense. Thanks.